### PR TITLE
Define biginteger, bigdecimal, byte, short, long, double for all implementations

### PR DESCRIPTION
**Pull request by aslakhellesoy**
# Description

Implement the numeric precision parameter types from the Java implementation in other implementations:

- [x] `biginteger`
- [x] `bigdecimal`
- [x] `byte`
- [x] `short`
- [x] `long`
- [x] `double`

# Motivation & context

In `@cucumber/language-service` (written in TypeScript) we need to construct a registry with types that match any implementation. If we don't do that, an error will be thrown when we encounter a parameter type such as `{double}` when we process Cucumber Expressions from Java. Therefore, all implementations must have the same built-in parameter types.

Moreover, this improves numeric type support for all implementations.

## Type of change

- New feature (non-breaking change which adds new behaviour)

### CODE REVIEW

File: go/parameter_type_registry.go [hunk 1]
```diff
@@ -74,7 +102,126 @@ func NewParameterTypeRegistry() *ParameterTypeRegistry {
...
+		false,
+	)
+	if err != nil {
+		panic(err)
+	}
+	err = result.DefineParameterType(bigdecimalParameterType)
+	if err != nil {
+		panic(err)
+	}
+
```
Comment (aurelien-reeves) [inline go/parameter_type_registry.go:224]: This looks like there is some pattern here
Is it possible to factorize it?

Maybe that would not be welcome or easy in go?

File: java/src/test/java/io/cucumber/cucumberexpressions/CucumberExpressionTest.java [hunk 2]
```diff
@@ -40,7 +40,8 @@
         List<Path> paths = new ArrayList<>();
         newDirectoryStream(Paths.get("..", "testdata", "cucumber-expression", "matching")).forEach(paths::add);
         paths.sort(Comparator.naturalOrder());
-        return paths;
+//        return paths;
```
Comment (aurelien-reeves) [inline java/src/test/java/io/cucumber/cucumberexpressions/CucumberExpressionTest.java:43]:
```suggestion
```

**Review by aurelien-reeves — COMMENTED** [review]
ts and ruby looks good
for jvm and go, other reviewers more confident with those languages would be welcome :D

File: go/parameter_type_registry.go (hunk 1 above)
Comment (aslakhellesoy) [reply go/parameter_type_registry.go:224]: I see what you mean but I think it would be clunky to refactor :-/

Comment (aurelien-reeves) [reply go/parameter_type_registry.go:224]: true

**Pull request merged**